### PR TITLE
Indexer: shovel config

### PR DESCRIPTION
**Pull request by psugihara**
Right now this just generates a shovel config that works to pull in all of the event data that the frontend uses.

The code in here only needs to run in dev so I used bun for easy typescript as they do in the shovel example. I copied the contracts stuff from index.js so that I could have a typed version (made writing this a lot easier).

Config generation happens in [shovel-config.ts](https://github.com/trifle-labs/anybody-problem/compare/indexer?expand=1#diff-f7729beb56e429c5cb1f3ae69272c8bd6d26ec34a7505948ebe8ec85869cb982). It is pretty easy to swap contracts and event names as you want to change.

If you want to try it locally...

```
brew install postgresql@16
brew services start postgresql@16
createdb shovel

curl -LO --silent https://indexsupply.net/bin/1.6/darwin/arm64/shovel
chmod +x shovel

./shovel -config config.json
```

⚒️⚒️⚒️⚒️⚒️
hot screenshots of my db...

Screenshot 2024-05-15 at 10 44 49 PM
Screenshot 2024-05-15 at 10 44 34 PM
Screenshot 2024-05-15 at 10 44 24 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,10 +23,13 @@
...
   },
   "lint-staged": {
-    "*.{js,jsx,ts,tsx}": [
+    "*.{js,jsx}": [
       "eslint --fix",
       "prettier --write --ignore-unknown"
     ],
+    "*.{ts,tsx}": [
+      "prettier --write --ignore-unknown"
+    ],
```
Comment (psugihara) [inline package.json:32]: eslint was choking on the ts, didn't really look into it

File: server/shovel-config.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+const mainnet: Source = {
+  name: 'mainnet',
+  chain_id: 1,
+  url: 'https://ethereum-rpc.publicnode.com'
+}
+
+const sepolia: Source = {
+  name: 'sepolia',
+  chain_id: 11155111,
+  url: 'https://rpc2.sepolia.org', // 'https://ethereum-sepolia-rpc.publicnode.com' rate limited
```
Comment (okwme) [inline server/shovel-config.ts:22]: we've got some paid endpoints as well in case these rate limit or time out

File: server/shovel-config.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        new ethers.Contract(contract.networks[network].address, abi)
+      ]
+    }
+  )
+)
+
+const solTypeToPgType = {
+  address: 'bytea',
+  uint256: 'numeric',
+  bytes32: 'bytea',
```
Comment (okwme) [inline server/shovel-config.ts:43]: would these be better as strings? never worked with `bytea` types in a database, are they tricky?

Comment (psugihara) [reply server/shovel-config.ts:43]: just tried using text type just now but it errors. I think this conversion is hardcoded in shovel (dumb that I have to specify it here 🤷)

Screenshot 2024-05-16 at 9 04 37 AM

from https://indexsupply.com/shovel/docs/


It looks like it's pretty easy to query though and postgres can `encode` and `decode` strings on the way in/out:

Screenshot 2024-05-16 at 9 10 13 AM

**Review by okwme — APPROVED** [review]
LGTM

**Pull request merged**